iml-gui/crate/src/components/command_modal.rs

```diff
@@ -572,12 +577,36 @@ fn extract_uri_id<T: EndpointName>(input: &str) -> Option<u32> {
     })
 }
 
-fn is_finished(cmd: &RichCommand) -> bool {
-    cmd.complete
+fn to_load_cmd(model: &Model, cmd_id: u32) -> bool {
+    // load it if the command is not found or is not complete
+    if let Some(cmd) = model.commands.get(&cmd_id) {
+        !cmd.complete
+    } else {
+        true
+    }
+}
+
+fn to_load_job(model: &Model, job_id: u32) -> bool {
+    if let Some(job) = model.jobs.get(&job_id) {
+        // job.state can be "tasked", "pending" or "complete"
+        // if a job is errored or cancelled, it is also complete
+        job.state != "complete"
+    } else {
+        true
+    }
```

Comment:
```suggestion
    model.jobs.get(&job_id).map(|j| j.state != "complete").unwrap_or(true)
```
Nit: shorter and linear ;-)

Reply:
It is already done in https://github.com/whamcloud/integrated-manager-for-lustre/pull/1899 as well :-)

Reply:
Ok, I thought it was déjà vu :)